src/build.rs

```diff
@@ -259,17 +259,58 @@ impl BuildQueue {
             cmd.env("CARGO_TARGET_DIR", &Path::new("target").join("rls"));
             cmd.current_dir(build_dir);
 
-            let mut new_cmd_line = match cmd.output() {
+            let new_cmd_line_args = match cmd.output() {
                 Ok(output) => {
                     let out = String::from_utf8(output.stderr).unwrap();
                     // println!("output: `{}`", out);
                     let exit_str = "     Running `";
+
                     match out.rfind(exit_str) {
                         Some(i) => {
                             let remaining = &out[i + exit_str.len() ..];
                             let end = remaining.find('`').unwrap();
-                            remaining[..end].to_owned()
+                            let remaining = &remaining[..end];
+                            let split = remaining.split(' ');
+
+                            let mut result = vec![];
+                            let mut in_quoted_arg = false;
+
+                            for element in split {
```

Comment:
Well, now we'd have to decide on an initial capacity for the buffer. Docs aren't clear about guaranteed behaviour, but atleast on playground the initial cap is 4 and doubles each time. So for a 20-char arg there'll be three re-allocations (8, 16, 32).